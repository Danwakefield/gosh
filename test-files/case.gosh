echo "9 Test cases"
case a in
	a) echo "SUCCESS 1"
	;;
	b) echo "FAIL 1"
	;;
esac

case a in
	(a) echo "SUCCESS 2"
	;;
	(b) echo "FAIL 2"
	;;
esac

case xyz in
	xyz) echo "SUCCESS 3"
	;;
	*) echo "FAIL 3"
esac

case xyz in
	*) echo "SUCCESS 4"
esac

case bb in
	[a-c][a-c]) echo "SUCCESS 5"
esac

case 1 in
	$((y=0)) ) echo "FAIL 6"
	;;
	$((y=1)) ) echo "SUCCESS 6"
esac

case a in
	x|y|z|a) echo "SUCCESS 7"
esac

V=abc
case abc in
	$V) echo "SUCCESS 8"
esac

case a in
	a) ;;
esac && echo "SUCCESS 9"
