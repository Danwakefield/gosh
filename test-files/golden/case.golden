9 Test cases
SUCCESS 1
SUCCESS 2
SUCCESS 3
SUCCESS 4
SUCCESS 5
SUCCESS 6
SUCCESS 7
SUCCESS 8
SUCCESS 9
